File: expense-report/csharp/src/ExpenseReport/ExpenseItem.cs
namespace ExpenseReport;

public class ExpenseItem
{
    public ExpenseItem(string description, Money amount, Category category)
    {
        Description = description;
        Amount = amount;
        Category = category;
    }

    public string Description { get; }
    public Money Amount { get; }
    public Category Category { get; }
    public bool IsOverLimit => Amount > SpendingPolicy.LimitFor(Category);
}

File: expense-report/typescript/src/ExpenseItem.ts
import { Category } from './Category.js';
import { Money } from './Money.js';
import { SpendingPolicy } from './SpendingPolicy.js';

export class ExpenseItem {
  constructor(
    public readonly description: string,
    public readonly amount: Money,
    public readonly category: Category,
  ) {}

  get isOverLimit(): boolean {
    return this.amount.greaterThan(SpendingPolicy.limitFor(this.category));
  }
}
